### ISSUE TITLE
False positive "proc macro not expanded" error when using any derive helpers

### ISSUE THREAD

**Issue by jonas-schievink** [rust-lang/rust-analyzer#12582]
```rust
#[derive(serde::Serialize)]
#[serde(rename = "abc")]
struct S;
```

This reports an error on the `#[serde]` attribute with "proc macro `serde` not expanded: proc macro not found", even though the code is valid.

**Comment by Veykril** [comment]
Ah this is probably caused by this branch https://github.com/rust-lang/rust-analyzer/blame/817f46bc6c36e5754eaf7f2ae4a8e96a1990291e/crates/hir-def/src/nameres/collector.rs#L1138-L1150

**Comment by Veykril** [comment]
Actually I can't reproduce this
